src/sage/schemes/elliptic_curves/ell_finite_field.py

```diff
@@ -2109,9 +2202,30 @@ def curves_with_j_0_char3(K):
     while not b or not b.trace():
         b = K.random_element()
 
-    if K.degree() % 2:
-        return [EllipticCurve(K, a4a6) for a4a6 in
-            [[1,0], [-1,0], [-1,b], [-1,-b]]]
+    deg = K.degree()
+    if deg % 2:
+        curves = [
+            EllipticCurve(K, a4a6)
+            for a4a6 in [[1, 0], [-1, 0], [-1, b], [-1, -b]]
+        ]
+
+        # It is a bit annoying here since we know the last two curves have order base +- delta, but
+        # we don't know the sign. I think you can compute it via some analysis on `b`, but that is
+        # slow, so I opt to use `.set_order` with check=True, so that if it fails it means it is the
+        # wrong order :)
```

Comment:
The `.set_order()` check can have false positives; is there a reason that can't happen here?

Reply:
I have some heuristics argument, but I think I will comment this out for now, and leave it for another issue (#37110)